scraper.rb

```diff
@@ -1,290 +1,46 @@
-        href = link['href']
-        # we can't do this as one operation as they don't always appear
-        # in the same order :(
-        term = href.to_s.match(/idLegislatura=(\d+)/).captures[0]
-        id = href.to_s.match(/idDiputado=(\d+)/).captures[0]
-        term_map[term.to_i] = id
-        save_membership_from_url(name, href)
-      end
-      if not term.nil? and legislature.has_css?('div.principal')
-        term_div = find('div.principal')
-        name, start_year, end_year = term_div.text.match(/(\w+\s*\w+)\s*\(\s*(\d+)\s*-\s*([^)]*)\)/).captures
-        if end_year.tidy == 'Actualidad'
-          end_year = ''
-        end
-        exists = ScraperWiki::select('id FROM terms WHERE id is ??', [id]) rescue nil
-        if exists.nil?
-          term = {
-            id: term,
-            name: name.tidy,
-            start_date: start_year.tidy,
-            end_date: end_year.tidy,
-            source: 'http://www.congreso.es/',
-          }
-          ScraperWiki.save_sqlite([:id], term, 'terms')
-        end
-      end
-    end
-  end
-
-  # the all terms page seems to be very unreliable so if we can't find what we expect
-  # then we should quite rather than trying to make up an incorrect ID
-  if term_map.empty?
-      return nil
-  end
-
-  min_term = term_map.keys.min
-
-  id = "#{min_term}_#{term_map[min_term]}"
-  for term in term_map.keys
-      ScraperWiki.sqliteexecute('update memberships set id = ? where id = 0 and term = ? and iddiputado = ?', [id, term, term_map[term]])
-  end
-  return id
-end
-
-def scrape_people(url)
-  visit url
-
-  all('div#RESULTADOS_DIPUTADOS div.listado_1 ul li a').each do |link|
-    save_membership_from_url(link.text, link['href'])
-  end
-
-  pagination = all('div.paginacion').first
-  next_page = nil
-  if pagination.has_xpath?(".//a[contains(.,'Página Siguiente')]")
-    within (pagination) do
-      next_page = find(:xpath, ".//a[contains(.,'Página Siguiente')]")
-    end
-  end
-
-  # the website is a bit fragile to lets not hammer it with requests
-  sleep(2)
-  unless next_page.nil?
-    scrape_people(next_page['href'])
+def person_data(data)
+  unwanted_keys = %i(party faction faction_id start_date end_date constituency term memberships_list)
+  data.reject do |k, _v|
+    unwanted_keys.include? k
   end
 end
 
-def scrape_memberships()
-  memberships = ScraperWiki::select('* FROM memberships')
-  for membership in memberships
-    scrape_person(membership['term'], membership['url'])
+member_list_url = 'http://www.congreso.es/portal/page/portal/Congreso/Congreso/Diputados/DiputadosTodasLegislaturas'
+merged_memberships = unmerged_memberships(member_list_url).flat_map do |mem|
+  other_memberships = mem[:memberships_list].memberships.reject{ |other_m| other_m.term == mem[:term] }
+  all_memberships = other_memberships.map do |other_m|
+    other_m.to_h.merge(person_data(mem))
   end
+  mem.delete(:memberships_list)
+  all_memberships.push mem
```

Comment:
I'm finding it very difficult to follow this logic, but as I suspect the whole approach is suspect due to the misplaced `memberships_list` field, I'm not sure it's worth digging too deeply at this point.